homeassistant/components/insteon/config_flow.py

```diff
@@ -0,0 +1,353 @@
+            }
+        )
+        return self.async_show_form(
+            step_id="init", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_plm(self, user_input=None):
+        """Set up the PLM modem type."""
+        errors = {}
+        if user_input is not None:
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_plm_schema(**schema_defaults)
+        return self.async_show_form(
+            step_id=STEP_PLM, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_hubv1(self, user_input=None):
+        """Set up the Hub v1 modem type."""
+        errors = {}
+        if user_input is not None:
+            user_input[CONF_HUB_VERSION] = 1
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            user_input.pop(CONF_HUB_VERSION)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_hub_v1_schema(**schema_defaults)
+        return self.async_show_form(
+            step_id=STEP_HUB_V1, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_hubv2(self, user_input=None):
+        """Set up the Hub v1 modem type."""
+        errors = {}
+        if user_input is not None:
+            user_input[CONF_HUB_VERSION] = 2
+            if await _async_connect(**user_input):
+                return self.async_create_entry(title="", data=user_input)
+            user_input.pop(CONF_HUB_VERSION)
+            errors["base"] = "cannot_connect"
+        schema_defaults = user_input if user_input is not None else {}
+        data_schema = build_hub_v2_schema(**schema_defaults)
+        return self.async_show_form(
+            step_id=STEP_HUB_V2, data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_import(self, import_info):
+        """Import a new connection as a config entry."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        if not await _async_connect(**import_info):
+            return self.async_abort(reason="cannot_connect")
+        return self.async_create_entry(title="", data=import_info)
+
+
+class InsteonOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle an Insteon options flow."""
+
+    def __init__(self, config_entry):
+        """Init the InsteonOptionsFlowHandler class."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Init the config flow."""
+        errors = {}
+        if user_input is not None:
+            change_hub_config = user_input.get(STEP_CHANGE_HUB_CONFIG, False)
```

Comment:
Maybe make this a dropdown with `vol.In` so they can only select one in the UI

Reply:
I had it that way originally but this approach allows me to use translations. For this specific selection, however, I am not sure that matters. Let me see how Insteon markets their product internationally and if they use different translations I will keep it this way.  If they use the English only version, I will do as you suggest.

Reply:
Looks like the modems are always referred to in English. So we go with your suggestion.